doc/source/whatsnew/v0.25.0.rst

```diff
@@ -358,6 +358,7 @@ Reshaping
 - :func:`to_records` now accepts dtypes to its `column_dtypes` parameter (:issue:`24895`)
 - Bug in :func:`concat` where order of ``OrderedDict`` (and ``dict`` in Python 3.6+) is not respected, when passed in as  ``objs`` argument (:issue:`21510`)
 - Bug in :func:`concat` where the resulting ``freq`` of two :class:`DatetimeIndex` with the same ``freq`` would be dropped (:issue:`3232`).
+- Bug in :func:`merge` where merging with equivalent Categorical dtypes was raising an error (:issue:`22501`)
```

Comment:
Does it make sense to include a whatsnew note for this since it was technically fixed in 0.24.0, albeit unknowingly?  No strong feelings either way.

Reply:
I think we've started to list issues being fixed as soon as we close them out, even if they might have been patched beforehand unknowingly, so I'd leave it here.